internal/ui/ui_glfw.go

```diff
@@ -200,38 +196,102 @@ func initialize() error {
 	return nil
 }
 
-type monitor struct {
+func setInitMonitor(m *glfw.Monitor) {
+	theUI.initMonitor = m
+	theUI.initDeviceScaleFactor = theUI.deviceScaleFactor(m)
+	// GetVideoMode must be called from the main thread, then call this here and record
+	// initFullscreen{Width,Height}InDIP.
+	v := m.GetVideoMode()
+	theUI.initFullscreenWidthInDIP = int(theUI.dipFromGLFWMonitorPixel(float64(v.Width), m))
+	theUI.initFullscreenHeightInDIP = int(theUI.dipFromGLFWMonitorPixel(float64(v.Height), m))
+}
+
+// Monitor is a wrapper around glfw.Monitor.
+type Monitor struct {
 	m  *glfw.Monitor
 	vm *glfw.VidMode
 	// Pos of monitor in virtual coords
-	x int
-	y int
+	x      int
+	y      int
+	width  int
+	height int
+	id     int
+	name   string
+}
+
+// Bounds returns the monitor's bounds.
+func (m *Monitor) Bounds() image.Rectangle {
+	return image.Rect(m.x, m.y, m.x+m.width, m.y+m.height)
+}
+
+// Name returns the monitor's name.
+func (m *Monitor) Name() string {
+	return m.name
 }
 
 // monitors is the monitor list cache for desktop glfw compile targets.
 // populated by 'updateMonitors' which is called on init and every
 // monitor config change event.
 //
 // monitors must be manipulated on the main thread.
-var monitors []*monitor
+var monitors []*Monitor
+
+// AppendMonitors appends the current monitors to the passed in mons slice and returns it.
+func (u *userInterfaceImpl) AppendMonitors(mons []*Monitor) []*Monitor {
+	u.m.RLock()
+	defer u.m.RUnlock()
+	return append(mons, monitors...)
+}
+
+// Monitor returns the window's current monitor. Returns nil if there is no current monitor yet.
+func (u *userInterfaceImpl) Monitor() *Monitor {
+	u.m.RLock()
+	defer u.m.RUnlock()
+	var glfwMonitor *glfw.Monitor
+	if u.isFullscreen() {
+		glfwMonitor = u.window.GetMonitor()
+	} else {
+		glfwMonitor = u.currentMonitor()
+	}
+	if glfwMonitor == nil {
+		return nil
+	}
+	for _, m := range monitors {
+		if m.m == glfwMonitor {
+			return m
+		}
+	}
+	monitor := glfwMonitorToMonitor(glfwMonitor)
+	return &monitor
```

Comment:
Monitor() returns a different address for each call, which we don't expect.

Reply:
We could return nil there, but I wasn't sure. It is only traveled if there is a `glfwMonitor` and `monitors` is empty or does not have any matching monitor for the `glfwMonitor` var.

The underlying problem is that `u.window.GetMonitor()` leads to calling `glfw.Window.GetMonitor()` which then returns `&Monitor{m}`. This means then when fullscreen, `u.window.GetMonitor()` always returns a new address.

----

As an aside, looking into the code, I should be able to change
```golang
	if u.isFullscreen() {
		glfwMonitor = u.window.GetMonitor()
	} else {
		glfwMonitor = u.currentMonitor()
	}
```

to

```golang
		glfwMonitor = u.currentMonitor()
```

Reply:
> It is only traveled if there is a glfwMonitor and monitors is empty or does not have any matching monitor for the glfwMonitor var.

I'm not sure this is an actual case. Returning nil might be fine.

Reply:
Any updates?

Reply:
Busy week.

I'll look into figuring out a solution for `u.window.GetMonitor()` returning new addresses this weekend. As I recall, it might be an internal glfw problem.